### PR TITLE
UX: Show fix message

### PR DESCRIPTION
**Pull request by MichaelA-ops**

### CODE REVIEW

**Comment by cmeiklejohn** [comment]
Care to add a few screenshots here just so we can see it?

**Comment by cmeiklejohn** [comment]
I'm assuming all of the error messages are not set correctly yet or I should assume that?

File: src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcStubbedRPCHasNoAssertionsException.java [hunk 1]
```diff
@@ -18,4 +18,24 @@ private static String getErrorMessage(String key) {
...
+                        "<a target=\"_blank\" href=\"https://filibuster-testing.github.io/filibuster-java-instrumentation/javadoc/cloud/filibuster/junit/statem/GrpcMock.html#verifyThat(io.grpc.MethodDescriptor,int)\">" +
+                            "Set an expectation that a stub will be invoked a particular number of times." +
+                        "</a>" +
+                    "</li>" +
+                    "<li>" +
+                        "<a target=\"_blank\" href=\"https://filibuster-testing.github.io/filibuster-java-instrumentation/javadoc/cloud/filibuster/junit/statem/GrpcMock.html#verifyThat(io.grpc.MethodDescriptor,ReqT,int)\">" +
+                            "Set an expectation that a stub will be invoked, with a given request, a particular number of times." +
+                        "</a>" +
+                    "</li>" +
+                    "<li>" +
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcStubbedRPCHasNoAssertionsException.java:35]: Third one shouldn't be here.  Should just be the verifyThat variants (2, I think.)

File: src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcThrownExceptionHasUnspecifiedFailureBehaviorException.java [hunk 2]
```diff
@@ -18,4 +18,15 @@ private static String getErrorMessage() {
         return "Test threw an exception, but no specification of failure behavior present. \nUse assertFaultThrows(...) " +
                 "to specify failure is expected when fault injected on this method, request or code.";
     }
+
+    @Override
+    public String getFixMessage() {
+        return "<ul>" +
+                    "<li>" +
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcThrownExceptionHasUnspecifiedFailureBehaviorException.java:25]: Should be all 4? variants of assertFaultThrows linked here.

Comment (cmeiklejohn) [reply src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcThrownExceptionHasUnspecifiedFailureBehaviorException.java:25]: I think there's actually more, it should link to all variants of asssertFaultThrows.

File: src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcSuppressedStatusCodeException.java [hunk 3]
```diff
@@ -18,4 +18,15 @@ private static String getErrorMessage() {
         return "Injected fault's status code was suppressed, but test indicates this should propagate directly upstream. " +
                 "\nEnsure that use of assertFaultPropagates(...) is correct.";
     }
+
+    @Override
+    public String getFixMessage() {
+        return "<ul>" +
+                    "<li>" +
+                        "<a target=\"_blank\" href=\"https://filibuster-testing.github.io/filibuster-java-instrumentation/javadoc/cloud/filibuster/junit/statem/FilibusterGrpcTest.html#assertTestBlock()\">" +
+                            "Place test assertions in assertTestBlock instead." +
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcSuppressedStatusCodeException.java:27]: assertFaultPropagates() variants.

File: src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcAmbiguousThrowAndErrorPropagationException.java [hunk 4]
```diff
@@ -18,4 +18,15 @@ private static String getErrorMessage() {
         return "Test indicates both throw and error propagation: too ambiguous.\n" +
                "Please verify you are only using either assertOnException(...) or assertFaultPropagates(...).";
     }
+
+    @Override
+    public String getFixMessage() {
+        return "<ul>" +
+                    "<li>" +
+                        "<a target=\"_blank\" href=\"https://filibuster-testing.github.io/filibuster-java-instrumentation/javadoc/cloud/filibuster/junit/statem/FilibusterGrpcTest.html#assertTestBlock()\">" +
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcAmbiguousThrowAndErrorPropagationException.java:26]: Should ask the user to remove an assertOnException or assertFaultPropagates as two different bullet points linking to each

Comment (cmeiklejohn) [reply src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcAmbiguousThrowAndErrorPropagationException.java:26]: like Remove a X, Remove a Y

File: src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcReadOnlyRPCUsedOutsideAssertOnExceptionException.java [hunk 5]
```diff
@@ -18,4 +18,15 @@ private static String getErrorMessage() {
         return "Use of readOnlyRPC not allowed outside of assertOnException(...) block. " +
                 "\nPlease rewrite code to specify precise assertions on mock invocations.";
     }
+
+    @Override
+    public String getFixMessage() {
+        return "<ul>" +
+                    "<li>" +
+                        "<a target=\"_blank\" href=\"https://filibuster-testing.github.io/filibuster-java-instrumentation/javadoc/cloud/filibuster/junit/statem/FilibusterGrpcTest.html#assertTestBlock()\">" +
+                            "Potentially place assertions in assertTestBlock instead of assertOnException." +
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcReadOnlyRPCUsedOutsideAssertOnExceptionException.java:27]: just link to assertOnException variants and readOnlyRPC variants I guess?

File: src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcSideEffectingRPCUsedOutsideAssertOnExceptionException.java [hunk 6]
```diff
@@ -18,4 +18,15 @@ private static String getErrorMessage() {
         return "Use of sideEffectingRPC not allowed outside of assertOnException(...) block. " +
                 "\nPlease rewrite code to specify precise assertions on mock invocations.";
     }
+
+    @Override
+    public String getFixMessage() {
+        return "<ul>" +
+                    "<li>" +
+                        "<a target=\"_blank\" href=\"https://filibuster-testing.github.io/filibuster-java-instrumentation/javadoc/cloud/filibuster/junit/statem/FilibusterGrpcTest.html#assertTestBlock()\">" +
+                            "Potentially place assertions in assertTestBlock instead of assertOnException." +
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcSideEffectingRPCUsedOutsideAssertOnExceptionException.java:27]: just link to assertOnException variants and sideEffectingRPC variants I guess?

File: src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcMultipleFaultsInjectedException.java [hunk 7]
```diff
@@ -17,4 +17,15 @@ public FilibusterGrpcMultipleFaultsInjectedException(Throwable cause) {
     private static String getErrorMessage() {
         return "Assertions in assertTestBlock failed due to multiple faults being injected. \nPlease use assertOnFault to update assertions so that they hold under fault.";
     }
+
+    @Override
+    public String getFixMessage() {
+        return "<ul>" +
+                    "<li>" +
+                        "<a target=\"_blank\" href=\"https://filibuster-testing.github.io/filibuster-java-instrumentation/javadoc/cloud/filibuster/junit/statem/FilibusterGrpcTest.html#assertTestBlock()\">" +
+                            "Place test assertion in assertTestBlock." +
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcMultipleFaultsInjectedException.java:26]: link to assertOnFault variants

**Comment by MichaelA-ops** [comment]
Screenshot 2023-07-19 at 02 35 53
Screenshot 2023-07-19 at 02 38 45

**Review by cmeiklejohn — CHANGES_REQUESTED** [review]
changelog = version 1.9.10

**Comment by cmeiklejohn** [comment]
I want to make some changes to these message, I'm not 100% on them yet.

File: src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcAmbiguousFailureHandlingException.java [hunk 8]
```diff
@@ -17,6 +17,17 @@ public FilibusterGrpcAmbiguousFailureHandlingException(Throwable cause) {
...
-                "\nPlease write an assertOnFaults(...) for this fault combination with appropriate assertions.";
+                "\nPlease write an assertOnFault(...) for this fault combination with appropriate assertions.";
+    }
+
+    @Override
+    public String getFixMessage() {
+        return "<ul>" +
+                    "<li>" +
+                        "<a target=\"_blank\" href=\"https://filibuster-testing.github.io/filibuster-java-instrumentation/javadoc/cloud/filibuster/junit/statem/FilibusterGrpcTest.html#assertOnFault(cloud.filibuster.junit.statem.CompositeFaultSpecification,java.lang.Runnable)\">" +
+                            "Write an assertion for the current fault combination using assertOnFault(CompositeFaultSpecification compositeFaultSpecification, java.lang.Runnable runnable)." +
```
Comment (cmeiklejohn) [inline src/main/java/cloud/filibuster/exceptions/filibuster/FilibusterGrpcTestRuntimeException/FilibusterGrpcAmbiguousFailureHandlingException.java:28]: Can we reformat the messages so it reads like 

"Write an assertion for the current fault combination using <LINK>assertOnFault(CompositeFaultSpecification compositeFaultSpecification, java.lang.Runnable runnable)</LINK>"

Having the entire thing a link is too difficult on the eyes.

**Pull request merged**